python/ctsm/crop_calendars/regrid_ggcmi_shdates.py

```diff
@@ -32,49 +42,95 @@ def define_arguments(parser):
     if not os.path.exists(regrid_output_directory):
         os.makedirs(regrid_output_directory)
 
     templatefile = os.path.join(regrid_output_directory, "template.nc")
-
-    # For some reason, doing ncks -v directly doesn't work. Have to copy it over first.
-    regrid_template_file = os.path.join(
-        regrid_output_directory,
-        os.path.basename(regrid_template_file_in),
-    )
-    shutil.copyfile(regrid_template_file_in, regrid_template_file)
-
     if os.path.exists(templatefile):
         os.remove(templatefile)
 
-    for v in ["planting_day", "maturity_day", "growing_season_length"]:
-        run_and_check(f"ncks -A -v area '{regrid_template_file}' '{templatefile}'")
-        run_and_check(f"ncrename -v area,{v} '{templatefile}'")
-    os.remove(regrid_template_file)
-    run_and_check(f"ncpdq -O -h -a -lat '{templatefile}' '{templatefile}'")
+    template_ds_in = xr.open_dataset(regrid_template_file_in)
+
+    # Process inputs
+    if crop_list is not None:
+        crop_list = crop_list.split(",")
+    if extension[0] != ".":
+        extension = "." + extension
+
+    # Import and format latitude
+    if "lat" in template_ds_in:
+        lat, Nlat = import_coord_1d(template_ds_in, "lat")
+    elif "LATIXY" in template_ds_in:
+        lat, Nlat = import_coord_2d(template_ds_in, "lat", "LATIXY")
+        lat.attrs["axis"] = "Y"
+    else:
+        abort("No latitude variable found in regrid template file")
+
+    # Flip latitude, if needed
+    if lat.values[0] < lat.values[1]:
+        lat = lat.reindex(lat=list(reversed(lat["lat"])))
+
+    # Import and format longitude
+    if "lon" in template_ds_in:
+        lon, Nlon = import_coord_1d(template_ds_in, "lon")
+    elif "LONGXY" in template_ds_in:
+        lon, Nlon = import_coord_2d(template_ds_in, "lon", "LONGXY")
+        lon.attrs["axis"] = "Y"
+    else:
+        abort("No longitude variable found in regrid template file")
+    template_da_out = xr.DataArray(
+        data=np.full((Nlat, Nlon), 0.0),
+        dims={"lat": lat, "lon": lon},
+        name="area",
+    )
+
+    # Save template Dataset for use by cdo
+    template_ds_out = xr.Dataset(
+        data_vars={
+            "planting_day": template_da_out,
+            "maturity_day": template_da_out,
+            "growing_season_length": template_da_out,
+        },
+        coords={"lat": lat, "lon": lon},
+    )
+    template_ds_out.to_netcdf(templatefile, mode="w")
 
-    input_files = glob.glob("*nc4")
+    # Loop through original crop calendar files, interpolating using cdo with nearest-neighbor
+    pattern = "*" + extension
+    input_files = glob.glob(pattern)
+    if len(input_files) == 0:
+        abort(f"No files found matching {os.path.join(os.getcwd(), pattern)}")
     input_files.sort()
     for f in input_files:
-        print(f[0:6])
+        this_crop = f[0:6]
+        if crop_list is not None and this_crop not in crop_list:
+            continue
+
+        print("    " + this_crop)
```

Comment:
Sam with this print as above about logging.

Reply:
Done with f39f7b4.